Snowflake.Data.Tests/UnitTests/Tools/UnixOperationsTest.cs

```diff
@@ -45,42 +47,76 @@ public void TestDetectGroupOrOthersWritablePermissions(
             var filePermissions = readWriteUserPermissions | groupOrOthersWritablePermissions | groupNotWritablePermissions | otherNotWritablePermissions;
             Syscall.chmod(filePath, filePermissions);
-            
+
             // act
             var result = s_unixOperations.CheckFileHasAnyOfPermissions(filePath, FileAccessPermissions.GroupWrite | FileAccessPermissions.OtherWrite);
-            
+
             // assert
             Assert.IsTrue(result);
         }
 
         [Test]
         public void TestDetectGroupOrOthersNotWritablePermissions(
-            [ValueSource(nameof(UserPermissions))] FilePermissions userPermissions, 
+            [ValueSource(nameof(UserPermissions))] FilePermissions userPermissions,
             [ValueSource(nameof(GroupNotWritablePermissions))] FilePermissions groupNotWritablePermissions,
             [ValueSource(nameof(OtherNotWritablePermissions))] FilePermissions otherNotWritablePermissions)
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
                 Assert.Ignore("skip test on Windows");
             }
-            
+
             var filePath = CreateConfigTempFile(s_workingDirectory, "random text");
             var filePermissions = userPermissions | groupNotWritablePermissions | otherNotWritablePermissions;
             Syscall.chmod(filePath, filePermissions);
-            
+
             // act
             var result = s_unixOperations.CheckFileHasAnyOfPermissions(filePath, FileAccessPermissions.GroupWrite | FileAccessPermissions.OtherWrite);
-            
+
             // assert
             Assert.IsFalse(result);
         }
 
+        [Test]
+        public void TestReadAllTextCheckingPermissions()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Assert.Ignore("skip test on Windows");
+            }
+            var content = "random text";
+            var filePath = CreateConfigTempFile(s_workingDirectory, content);
+            var filePermissions = FileAccessPermissions.UserReadWriteExecute;
+            Syscall.chmod(filePath, (FilePermissions)filePermissions);
+
+            // act
+            var result = s_unixOperations.ReadAllText(filePath, SnowflakeTomlConnectionBuilder.GetFileValidations());
+
+            // assert
+            Assert.AreEqual(content, result);
+        }
+
+        [Test]
+        public void TestShouldThrowExceptionIfOtherPermissionsIsSetWhenReadAllText()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Assert.Ignore("skip test on Windows");
+            }
+            var content = "random text";
+            var filePath = CreateConfigTempFile(s_workingDirectory, content);
+            var filePermissions = FileAccessPermissions.UserReadWriteExecute | FileAccessPermissions.OtherReadWriteExecute;
```

Comment:
It would be good if you would change the test to parametrised and test more combinations of permissions

Reply:
Added cases of using FileAccessPermissions that throws the exception only 2 cases.

Reply:
Added cases for all combinations with 600 permissions

Reply:
https://github.com/snowflakedb/snowflake-connector-net/blob/d5237fc0a9384260ea7da75997288f81fb99c6b0/Snowflake.Data.Tests/UnitTests/Tools/UnixOperationsTest.cs#L106-L122

Reply:
thanks, maybe we could just give more meaningful name for the test - because there is no information in the name that we test permissions on toml configuration. Maybe: TestFailIfGroupOrOthersHavePermissionsToTomlConfiguration?

Reply:
Changed test name to TestFailIfGroupOrOthersHavePermissionsToFileWithTomlConfigurationValidations. Thank you!!